app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,86 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      @username = current_user.username
+      erb :"tweets/tweets"
+    else
+      flash[:notice] = "You must be logged in to view this page."
+      redirect to "/login"
+    end
+  end
+
+  get "/tweets/new" do
+    if logged_in?
+      erb :"tweets/create_tweet"
+    else
+      flash[:notice] = "You must be logged in to create a tweet."
+      redirect to "/login"
+    end
+  end
+
+  get "/tweets/:id/edit" do
+    if !logged_in?
+      flash[:notice] = "You must be logged in to edit a tweet."
+      redirect to "/login"  
+    end
+    @tweet = Tweet.find(params[:id])
+    if @tweet && (@tweet.user_id == current_user.id)
+      erb :"tweets/edit_tweet"
+    else
+      flash[:notice] = "You do not have permission to edit tweets you did not create."
+      redirect to "/login"
+    end
+  end
+
+  post "/tweets" do
+    if params["content"].empty?
+      flash[:notice] = "You did not enter any thoughts to create a tweet."
+      redirect to "/tweets/new" 
+    end
+    @tweet = Tweet.new(content: params["content"], user_id:current_user.id)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.